code/obj/soup_pot.dm

```diff
@@ -93,6 +93,8 @@
 				return
 
 			else if ((istype(W, /obj/item/match) || istype(W, /obj/item/device/light/candle)) && W:on)
```

Comment:
any reason the check can't go up here?

Reply:
It already checks for `W:on`. But what it needs to do is check `W.on == 1` because the `on `variable in `obj/item/match` is 1 when it's on, and -1 when it's burnt out or broken. and if on is -1, then `W:on` evaluates to true.